mkosi

```diff
@@ -2266,76 +2298,75 @@ def install_arch(args: CommandLineArguments, root: str, do_run_build_script: boo
     # Set up system with packages from the base group
     run_pacstrap(packages)
 
-    if not do_run_build_script and args.bootable:
-        # Patch mkinitcpio configuration so:
-        # 1) we remove autodetect and
-        # 2) we add the modules needed for encrypt.
-        def jj(line: str) -> str:
-            if line.startswith("HOOKS="):
-                if args.encrypt == "all":
-                    return 'HOOKS="systemd modconf block sd-encrypt filesystems keyboard fsck"\n'
-                else:
-                    return 'HOOKS="systemd modconf block filesystems fsck"\n'
-            return line
-        patch_file(os.path.join(root, "etc/mkinitcpio.conf"), jj)
-
-        # Disable fallback images to save time. We override the preset files
-        # before installing any kernels to have the updated config available
-        # when the mkinitcpio pacman hook is run during installation.
-        presetdir = os.path.join(root, "etc/mkinitcpio.d")
-        for kernel in kernel_packages:
-            preset = os.path.join(presetdir, f"{kernel}.preset")
-            with open(preset, "w") as f:
-                f.write(
-                    dedent(
-                        f"""
-                        ALL_config="/etc/mkinitcpio.conf"
-                        ALL_kver="/boot/vmlinuz-{kernel}"
-                        PRESETS=('default')
-                        default_image="/boot/initramfs-{kernel}.img"
-                        """
-                    )
-                )
-
     # Install the user-specified packages and kernel
     packages = set(args.packages)
-    if not do_run_build_script and args.bootable:
-        packages |= kernel_packages
+    if not do_run_build_script and args.bootable and not any(["linux" in package for package in args.packages]):
```

Comment:
```suggestion
    if not do_run_build_script and args.bootable and not any("linux" in package for package in args.packages):
```
to not generate the whole list unnecessarily or
```suggestion
    if not do_run_build_script and args.bootable and not any(package.startswith("linux") for package in args.packages):
```
to better check that it's a kernel package as well, because otherwise `util-linux` would already trigger the `any`.

Thinking about this, I guess that the check against `official_kernel_packages` should maybe remain, because there are some packages that are not the kernel that have linux in their name or might even start with linux (e.g. linuxconsole), which could lead to images being created without a kernel.

Reply:
You're right, this is actually broken. I'll add back the official kernel packages.